components/typeahead/typeahead-container.component.ts

```diff
@@ -8,20 +8,20 @@ import { TypeaheadDirective } from './typeahead.directive';
 import { TypeaheadMatch } from './typeahead-match.class';
 
 const bs4 = `
-  <div class="dropdown-menu"
-       [ngStyle]="{top: top, left: left, display: 'block'}"
+  <div #ulElement class="dropdown-menu"
```

Comment:
I will make a component of each dom element we want to grab. So we don't need to use element referencing by hashtag. 
When the LI and UL element are angular 2 components I can just use the code below in typehead-container.component

`@ViewChild(TypeaheadListComponent) list: TypeaheadListComponent`
`@ViewChildren(TypeaheadListItemComponent) listItems: QueryList<TypeaheadListItemComponent>`

So I will create additional components just for the UL element (`TypeaheadListComponent`)  and a component for the LI elements (`TypeaheadListItemComponent`). This way I can grab the dom elements without hashtag. What do you think about that?